### ISSUE TITLE
Container exits immediately when using JDWP in JAVA_OPTIONS

### ISSUE THREAD

**Issue by joakime** [eclipse/jetty.docker#12]
<a href="https://github.com/milo2048"><img src="https://avatars0.githubusercontent.com/u/32928740?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [milo2048](https://github.com/milo2048)**
_Wednesday Jan 22, 2020 at 16:26 GMT_
_Originally opened as https://github.com/appropriate/docker-jetty/issues/130_

----

Running with docker-compose, using jetty:9.4-jre8, when I specify this:
```
environment:
     JAVA_OPTIONS=-Xmx2048m -Xms2048m -XX:MaxPermSize=512m -agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005

```
The Jetty container exits immediately with code 0.  Removing the -agentlib causes it to run normally.

**Comment by joakime** [comment]
<a href="https://github.com/milo2048"><img src="https://avatars0.githubusercontent.com/u/32928740?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [milo2048](https://github.com/milo2048)**
_Wednesday Jan 22, 2020 at 16:27 GMT_

----

I checked out the latest code and rebuilt the image using docker-entrypoint.sh from commit 6938dbcf3c3f1e433fca63909400670a52f39dfe but otherwise using the code in master.  The new image worked as expected.  This leads me to believe the issue is with the latest entrypoint script.

**Comment by joakime** [comment]
<a href="https://github.com/ngvtsn"><img src="https://avatars0.githubusercontent.com/u/804180?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [ngvtsn](https://github.com/ngvtsn)**
_Wednesday Mar 11, 2020 at 12:20 GMT_

----

Same. Looks like `docker-entrypoint.sh --dry-run` generates wrong jetty.start

**Comment by joakime** [comment]
<a href="https://github.com/sittingbool"><img src="https://avatars1.githubusercontent.com/u/12949701?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [sittingbool](https://github.com/sittingbool)**
_Friday Apr 24, 2020 at 14:08 GMT_

----

Same here, how can i resolve this?

**Comment by lachlan-roberts** [comment]
@milo2048 @ngvtsn @sittingbool

Looks like this specific issue is due to the fact that the remote debug outputs the string
```
Listening for transport dt_socket at address: 5005
```

I can see that we could use some improvement on the way the `jetty.start` file is generated. 
But in the meantime you should be able to work around this issue by using the `quiet=y` option:
```Dockerfile
ENV JAVA_OPTIONS='-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005,quiet=y'
```

This stops the `Listening for transport ....` string from being output and breaking the `jetty.start` file.

**Comment by lachlan-roberts** [comment]
This has been fixed with PR #27 and will be available with a `9.4.32` release
